Modified pm diffussion

ChamaELK opened the pull request.

Commit:
recently added diffusion content

Commit:
diffusion process with cython

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,160 @@
+# cython: infer_types=True
+
+import PIL
+import math
```
PIL and math are unused, you can remove them

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,160 @@
+        return nu, changed
+
+def setup_logging(format='%(asctime)-15s: %(message)s', level=logging.INFO):
+    logging.basicConfig(format=format, level=level)
```
You should remove this setup_logging helper

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,160 @@
+logger = logging.getLogger(__name__)
+
+class DiffusionRop(PerChannelRop):
+    def __init__(self,raw=None,T=0.0025,L=1.0/180,R=2,thr=3,steps=20,dt=1.0/7):
```
Try to follor [PEP8 style](https://www.python.org/dev/peps/pep-0008/)

that means:

``` python
def __init__(self, raw=None, T=0.0025, L=1.0/180, R=2, thr=3, steps=20, dt=1.0/7):
```

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,160 @@
+    def process_channel(self,data,detected=None): 
+        accelerated=False
+        denoiser = Denoise(self.L,accelerated)
+        return denoiser.modifiedPMdiffusion(data,self.steps,self.R,self.T,self.thr,self.dt)
```
Again, PEP8 style here sould mean spaces between arguments:

``` python
return denoiser.modifiedPMdiffusion(data, self.steps, self.R, self.T, self.thr, self.dt)
```

klaussfreire replied:
If `data` isn't of type `numpy.float32`, this code will break.

So you should forcibly cast it to `float32` to make sure:

``` python
if data.dtype.char != 'f':
    data = data.astype(numpy.float32)
```

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,160 @@
+    def modifiedPMdiffusion(self,noisy,steps, nradius,T,thr,dt):
+        ut=noisy
+        ssim = numpy.zeros(shape=steps, dtype=numpy.float32)
+        psnr = numpy.zeros(shape=steps, dtype=numpy.float32)
```
These are unused

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,160 @@
+            maskt, evolvedut= self.binarymask(ut,nradius,T,thr)
+            ut, changed = self.iteration(maskt,evolvedut,dt)
+            #psnr[i]= measure.compare_psnr(evolvedut,ut)
+            #ssim[i] = measure.compare_ssim(evolvedut,ut)
```
Remove commented code (if it's experimental code, you can re-add it later in your branch, but the PR should be free of commented code)

Commit:
Code cleanup

Remove unused imports, setup_logging, pep8 fixes, psnr ssim code,
casting data to float32

Commit:
pep8

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,154 @@
+logger = logging.getLogger(__name__)
+
+
+class DiffusionRop(PerChannelRop):
```
/ask what is this class meant to implement?

klaussfreire commented on cvastrophoto/rops/denoise/diffusion.py:
```diff
@@ -0,0 +1,154 @@
+            data, self.steps, self.R, self.T, self.thr, self.dt)
+ 
+     
+@cython.cfunc
+@cython.locals(gradient=cython.double, L=cython.double, g2=cython.double)
+@cython.returns(cython.double)
+@cython.nogil
+def g(L,gradient):
+    g2 = fabs(gradient)/L
+    return (1.0/(1.0+ g2*g2))
```
/ask "what is this function computing in the context of this module?"
